tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/TestBlockC.scala

```diff
@@ -0,0 +1,140 @@
+package it.unibo.scafi.test.functional.stdlib
+
+import it.unibo.scafi.test.FunctionalTestIncarnation._
+import it.unibo.scafi.test.functional.ScafiAssertions.assertNetworkValues
+import it.unibo.scafi.test.functional.ScafiTestUtils
+import org.scalatest._
+
+class TestBlockC extends FunSpec with BeforeAndAfterEach {
+  import ScafiTestUtils._
+
+  var net: Network with SimulatorOps = standardNetwork()
+  def restartNetwork(): Unit = {
+    net = standardNetwork()
+    net.addSensor(name = "source", value = false)
+  }
+  override protected def beforeEach(): Unit = restartNetwork()
+
+  val defaultNtimes: Int = someRounds
+
+  private[this] trait TestProgram extends AggregateProgram with StandardSensors with BlockC {
+    def hopGradient(source: Boolean): Double =
+      rep(Double.PositiveInfinity) { hops => {mux(source) {0.0} {1 + minHood(nbr {hops})}}}
+  }
+
+
+  describe("BlockC") {
+    describe("should support smaller") {
+      it("should work in basic scenario - all true") {
+        exec(new TestProgram {
+          override def main(): Boolean = smaller(1, 2)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          true, true, true,
+          true, true, true,
+          true, true, true
+        )).toMap)(net)
+      }
+      it("should work in basic scenario - all false") {
+        exec(new TestProgram {
+          override def main(): Boolean = smaller(2, 1)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          false, false, false,
+          false, false, false,
+          false, false, false
+        )).toMap)(net)
+      }
+      it("should work with nbr operations") {
+        exec(new TestProgram {
+          override def main(): Boolean = smaller(mid(), minHoodPlus(nbr(mid())))
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          true, false, false,
+          false, false, false,
+          false, false, true
+        )).toMap)(net)
+      }
+    }
+    describe("should support findParent") {
+      it("should work with a constant value") {
+        exec(new TestProgram {
+          override def main(): ID = findParent(1)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          Int.MaxValue, Int.MaxValue, Int.MaxValue,
+          Int.MaxValue, Int.MaxValue, Int.MaxValue,
+          Int.MaxValue, Int.MaxValue, Int.MaxValue
+        )).toMap)(net)
+      }
+      it("should work with mid") {
```

Comment:
The use of `mid` is not very clear to an external reader: I would add a comment specifying that `mid` induces a gradient field
```
(0  1  2
 3  4  5
 6  7  8)
```